dart/sign_in_with_google/lib/main.dart

```diff
@@ -0,0 +1,120 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:dart_appwrite/dart_appwrite.dart';
+import 'package:dart_appwrite/models.dart';
+import 'package:http/http.dart' as http;
+
+Future<dynamic> main(final context) async {
+  final requiredEnvVars = ['GOOGLE_CLIENT_ID'];
+  for (var varName in requiredEnvVars) {
+    if (Platform.environment[varName]?.isEmpty ?? true) {
+      throw Exception('Environment variable $varName must be set.');
+    }
+  }
+
+  final googleClientId = Platform.environment['GOOGLE_CLIENT_ID']!;
+
+  final reqBody = context.req.bodyJson as Map<String, dynamic>;
+  final idToken = reqBody['idToken'] ?? '';
+
+  // Validate input
+  if (idToken.isEmpty) {
+    throw Exception('idToken must be provided in the request body.');
+  }
+
+  // Verify the Google ID token
+  final tokenInfoResponse = await http.get(
+    Uri.parse('https://oauth2.googleapis.com/tokeninfo?id_token=$idToken'),
+  );
+
+  if (tokenInfoResponse.statusCode != 200) {
+    throw Exception(
+      'Failed to verify Google ID token: ${tokenInfoResponse.body}',
+    );
+  }
+
+  final tokenInfo = json.decode(tokenInfoResponse.body);
+
+  // Verify the token's audience matches our client ID
+  final aud = tokenInfo['aud'] ?? '';
+  if (aud != googleClientId) {
+    throw Exception('ID Token audience does not match the expected client ID.');
+  }
+
+  // Verify the token is issued by Google
+  final iss = tokenInfo['iss'] ?? '';
+  if (iss != 'https://accounts.google.com' && iss != 'accounts.google.com') {
+    throw Exception('ID Token is not issued by Google.');
+  }
+
+  // Verify the token has not expired
+  final exp = int.tryParse(tokenInfo['exp']?.toString() ?? '0') ?? 0;
+  final currentTime = DateTime.now().millisecondsSinceEpoch ~/ 1000;
+  if (exp <= currentTime) {
+    throw Exception('ID Token has expired.');
+  }
+
```

Comment:
@ahmtydn this is missing validation of the signature as described [here](https://developers.google.com/identity/gsi/web/guides/verify-google-id-token#:~:text=The%20ID%20token%20is%20properly%20signed%20by%20Google.%20Use%20Google%27s%20public%20keys%20(available%20in%20JWK%20or%20PEM%20format)%20to%20verify%20the%20token%27s%20signature.%20These%20keys%20are%20regularly%20rotated%3B%20examine%20the%20Cache%2DControl%20header%20in%20the%20response%20to%20determine%20when%20you%20should%20retrieve%20them%20again.).